tmva/tmva/src/MethodDL.cxx

```diff
@@ -340,6 +344,8 @@ void MethodDL::ProcessOptions()
          settings.regularization = DNN::ERegularization::kL1;
       } else if (regularization == "L2") {
          settings.regularization = DNN::ERegularization::kL2;
+      } else {
```

Comment:
This is fine, but a change to consider is making a separate option for each regulariser (in principle one can use them at the same time).
E.g. `L2_Regularisation=1.5:L1_Regularisation=0.`. The default of =0. would disable the regulariser.
Right now it is unclear to me how you control the strength of the regularisation. (Is this `WeightDecay`?)

Reply:
yes weight decay controls the strength !
I am not sure I understand your comment about  L2_Regularisation=1.5:L1_Regularisation=0..
Now you have a string defining the regularisation, can be "None","L1","L2"

Reply:
I'm just saying that in e.g. xgboost the configuration options are
`xgb.train(..., l2_reg=1., l1_reg=0.2, ...)`, that is they are separate, and maybe we would want to have a similar interface.